### ISSUE TITLE
Autosize not adjusting after its value is $set from a hintAction

### ISSUE THREAD

**Issue by davidwhicker** [laravel-filament/filament#13879]
### Package

filament/filament

### Package Version

v3.2.98

### Laravel Version

v11.19

### Livewire Version

v3

### PHP Version

PHP 8.3

### Problem description

Autosize is not adjusting its size after a hintAction WITH a form has $set its value. 

`Textarea::make('collectFrom')->autosize()->rows(5)->hintActions(
                            [
                                \Filament\Forms\Components\Actions\Action::make('collect_client_address_modal')
                                    ->label('Client')
                                    ->icon('heroicon-o-magnifying-glass')
                                    ->modalSubmitActionLabel('Add Address')
                                    ->form(fn(Get $get)=>self::searchClientActionSchema($get))
                                    ->action(fn (array $data, Set $set) => $set('collectFrom', Client::find($data['client_id'])->address ?? null)),
                            ]
                        ),`
                        
If I skip the ->form section and apply only a custom action() eg >action(fn (array $data, Set $set) => $set('collectFrom', Client::find(1072)->address ?? null)), then it will work. 

In short it seems if the hint action requires a modal and the submit action is to $set a new value in the Textarea then it will not update to its new size. 

To confirm, the textfield is not collaposing like the preivous release, its simply not updating to its new size ( IF a model is in the hintaction )

### Expected behavior

Textarea to adjust to new size

### Steps to reproduce

as above

### Reproduction repository (issue will be closed if this is not valid)

REally sorry but I cannot get one live. It should be very easy to reproduce from the PR https://github.com/filamentphp/filament/pull/13748
